subprojects/docs/src/docs/userguide/moreAboutTasks.adoc

```diff
@@ -593,40 +593,45 @@ This ensures that changes to the implementation of a nested input causes the bui
 By this it is also possible to add user provided code as an input, e.g. by annotating an `@api:org.gradle.api.Action[]` property with `@api:org.gradle.api.tasks.Nested[]`.
 Note that any inputs to such actions should be tracked, either by annotated properties on the action or by manually registering them with the task.
 
-Using nested inputs allows richer modeling and extensibility for tasks, as e.g. shown by `api:org.gradle.api.tasks.compile.CompileOptions#getCompilerArgumentProviders[]`.
+Using nested inputs allows richer modeling and extensibility for tasks, as e.g. shown by `api:org.gradle.api.tasks.testing.Test#getJvmArgumentProviders[]`.
 
-For example, to declare annotation processor arguments, it is possible to do the following:
+This allows us to https://github.com/gradle/gradle/blob/7b047c7cbb4932743243a76123f5347be6d07856/subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPluginExtension.java#L138-L157[model the Jacoco Java agent] and add the corresponding inputs and JVM arguments in one go:
```

Comment:
`Jacoco` -> `JaCoCo`

Reply:
`in one go`: that's not the reason why we want this, but instead because we can provide the necessary metadata for each element and let Gradle treat them as the inputs they are.

Reply:
Changed the wording.